Expensify/App
# Merge TextInputWithPrefix into TextInput

### Details

- [x] TextInputWithPrefix

### Fixed Issues

$ https://github.com/Expensify/App/issues/6584
$ https://github.com/Expensify/App/issues/7308

### Tests



### PR Review Checklist

#### Contributor (PR Author) Checklist

#### PR Reviewer Checklist

### QA Steps
1. Open NewRoom page on All plaforms. 
2. Check that you are able to operate the Prefixed new room input. 
3. Go to ReportSettingsPage and try to rename. 
4. Check a few other pages which as normal text input and observe that design on those input should not change.


### Tested On


### Screenshots


#### Web | Desktop

![image](https://user-images.githubusercontent.com/24370807/156064420-33046ed6-a281-4d84-8fb8-43a1401717be.png)
![image](https://user-images.githubusercontent.com/24370807/156201792-384fa9e7-e7bc-4ac4-a59d-4ff18c450995.png)


#### Mobile Web

![image](https://user-images.githubusercontent.com/24370807/156202918-810343e9-9ad6-4448-af63-89003e3d6152.png)


#### iOS

![image](https://user-images.githubusercontent.com/24370807/156210069-eaa4004e-d24f-4f63-a61c-33ccb136576c.png)

#### Android

![image](https://user-images.githubusercontent.com/24370807/156064899-ba51d90e-8774-4acf-ab9b-ea7bd028e353.png)
![image](https://user-images.githubusercontent.com/24370807/156201725-55af9c2e-57e7-4859-94df-88688d763fd6.png)
